airflow/providers/openai/hooks/openai.py

```diff
@@ -0,0 +1,89 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from __future__ import annotations
+
+from typing import Any
+
+import openai
+
+from airflow.hooks.base import BaseHook
+
+
+class OpenAIHook(BaseHook):
+    """
+    Use OpenAI SDK to interact with OpenAI APIs.
+
+    .. seealso:: https://platform.openai.com/docs/introduction/overview
+
+    :param conn_id: :ref:`Cohere connection id <howto/connection:openai>`
+    """
+
+    conn_name_attr = "conn_id"
+    default_conn_name = "openai_default"
+    conn_type = "openai"
+    hook_name = "OpenAI"
+
+    def __init__(self, conn_id: str = default_conn_name, *args: Any, **kwargs: Any) -> None:
+        super().__init__(*args, **kwargs)
+        self.conn_id = conn_id
+        openai.api_key = self._get_api_key()
```

Comment:
I don't look other code yet but want to mention that current `openai` library implementation doesn't work well in multiple threads: https://github.com/openai/openai-python/discussions/631

So maybe we should wait a bit and wait until [1.0.0 release](https://github.com/openai/openai-python/discussions/631) and fix this issue, as well as introduce breaking changes

There are also some other issues currently exists, which I hope fixed into the new major version of library.

Reply:
@Taragolis I see there are a bunch of issues, what if release a minor/patch version of this provider as well? Just to indicate this is not production-ready?

cc: @kaxil @mpgreg

Reply:
@Taragolis I see there are a bunch of issues, what if release a minor/patch version of this provider as well? Just to indicate this is not production-ready?

cc: @kaxil @mpgreg

Reply:
Yes let's get it out with a note on status.

Reply:
And maybe also better to pin openai library <1.0

Reply:
Made the required changes, please check.